### PR TITLE
added bytes conversion

### PR DESCRIPTION
**Pull request by ChristianMertes**
## What do these changes do?

Added `__bytes__()` magic method so that `bytes(url)` will work and use optimal ASCII encoding, as URL string representations are always encoded already.

While I was at it, I fixed some typos I saw and I had to add a mypy ignore for the tests to work for me. There is one test that fails for me (`test_semicolon_as_separator`) because I was using a Python > 3.9.2 and `parse_qsl()` [changed its behavior](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qsl).

## Are there changes in behavior for the user?

Existing functionality shouldn't be affected.

## Related issue number

n/a

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by ChristianMertes** [comment]
It is interesting that said test also fails for the other Python versions. I am not convinced this has anything to do with the proposed changes, however....

File: yarl/_url.py [hunk 1]
```diff
@@ -272,6 +272,9 @@ def __str__(self):
     def __repr__(self):
         return "{}('{}')".format(self.__class__.__name__, str(self))
 
+    def __bytes__(self):
+        return str(self).encode("ascii")
```
Comment (MyrikLD) [inline yarl/_url.py:276]: Why not UTF-8?

Comment (ChristianMertes) [reply yarl/_url.py:276]: It's been a while but IIRC the string representation already didn't use any non-ascii characters as everything is URL-encoded, right? Or can there be Unicode in the domain name?

Comment (asvetlov) [reply yarl/_url.py:276]: No, unicode domains are idna encoded, e.g. `xn--h1aagokeh.xn--p1ai`

File: setup.cfg [hunk 2]
```diff
@@ -30,3 +30,5 @@ ignore_missing_imports = true
 [mypy-pytest]
 ignore_missing_imports = true
 
+[mypy-multidict]
+ignore_missing_imports = true
```
Comment (asvetlov) [inline setup.cfg:34]:
```suggestion
```

multidict *provides* type annotations

**Comment by asvetlov** [comment]
Suggested typos are applied to master in a separate commit, thanks

**Review by asvetlov — APPROVED** [review]

**Pull request merged**